db/system_keyspace.cc

```diff
@@ -2642,6 +2655,304 @@ class clients_table : public streaming_virtual_table {
+            boost::sort(range._endpoints);
+
+            auto token_end = data_value(range._end_token);
+            auto endpoints = boost::copy_range<std::vector<data_value>>(range._endpoints | boost::adaptors::transformed([] (const auto& endpoint) {
+                return data_value(endpoint);
+            }));
+            auto endpoints_list = make_list_value(list_type, endpoints);
+
+            return std::pair(token_end, endpoints_list);
+        }));
+
+        co_return make_map_value(map_type, map_type_impl::native_type(
+            std::make_move_iterator(ring_ranges.begin()),
+            std::make_move_iterator(ring_ranges.end())
+        ));
+    }
+
+    future<> execute(reader_permit permit, result_collector& result, const query_restrictions& qr) override {
+        using desc = cql3::statements::describe_statement;
+        auto cluster = _db.get_config().cluster_name();
+        auto partitioner = _gossiper.get_partitioner_name();
+        auto snitch = _db.get_snitch_name();
+
+        // General entry
+        auto general_dk = make_partition_key("");
+        if (this_shard_owns(general_dk) && contains_key(qr.partition_range(), general_dk)) {
+            co_await result.emit_partition_start(general_dk);
+            clustering_row general_row(clustering_key::make_empty());
+            set_cell(general_row.cells(), to_bytes(desc::CLUSTER_COLUMN), cluster);
+            set_cell(general_row.cells(), to_bytes(desc::PARTITIONER_COLUMN), partitioner);
+            set_cell(general_row.cells(), to_bytes(desc::SNITCH_COLUMN), snitch);
+
+            co_await result.emit_row(std::move(general_row));
+            co_await result.emit_partition_end();
+        }
+
+        // Entries for user keyspaces
+        auto user_keyspaces = _db.get_user_keyspaces();
+        for (auto& ks_name : user_keyspaces) {
+            auto dk = make_partition_key(ks_name);
+            if (!this_shard_owns(dk) || !contains_key(qr.partition_range(), dk)) {
+                continue;
+            }
+
+            auto map_value = co_await get_range_to_address_data_value(ks_name);
+            co_await result.emit_partition_start(dk);
+            clustering_row cr(clustering_key::make_empty());
+            set_cell(cr.cells(), to_bytes(desc::CLUSTER_COLUMN), cluster);
+            set_cell(cr.cells(), to_bytes(desc::PARTITIONER_COLUMN), partitioner);
+            set_cell(cr.cells(), to_bytes(desc::SNITCH_COLUMN), snitch);
+            set_cell(cr.cells(), to_bytes(desc::RANGE_OWNERSHIP_COLUMN), map_value);
+
+            co_await result.emit_row(std::move(cr));
+            co_await result.emit_partition_end();
+        }
+    }
+
+public:
+    cluster_description_table(replica::database& db, service::storage_service& ss, gms::gossiper& gossiper) : streaming_virtual_table(build_schema()),
+        _db(db), _ss(ss), _gossiper(gossiper) {}
+};
+
+class elements_description_table : public streaming_virtual_table {
```

Comment:
Maybe `table_elements_description_table`. Currently it is completely unclear just what elements this is about.

Reply:
The name was unclear indeed. 
By elements I was referring to: keyspaces, tables, views, indexes, UDT, UDF and UDA, Since they all belong to some keyspace, I've renamed it to `keyspace_elements_description_table` (both class name and table name and I believe I've updated all related comments and commit messages).